src/js/containers/FormPage.jsx

```diff
@@ -49,36 +49,15 @@ class FormPage extends React.Component {
       newData = _.set([this.props.route.pageConfig.arrayPath, this.props.params.index], formData, this.props.form.data);
     }
     this.props.setData(newData);
-  }
-
-  onSubmit = ({ formData }) => {
```

Comment:
The `onChange` function in this component provides the same functionality as the `onSubmit` function except for the router push, which is exactly what we wanted to extract. Therefore, all I had to do was pull out the `onChange` and `goBack` functions (all `goBack` does is the router push), and provide a hook into the `onChange` function for the high-order component to know when it can push the next path (i.e., after the new data has been set).